penrose-web/src/inspector/views/mod/LabeledInput.tsx

```diff
@@ -67,24 +69,28 @@ class LabeledInput extends React.Component<IProps> {
     }
     public handleChange = (eattr: string, event: React.ChangeEvent<HTMLInputElement|HTMLSelectElement>) => {
         const teval = (isNaN(Number(event.target.value))) ? event.target.value : +event.target.value; // convert to number if necessary
-        this.updateAttr(eattr, teval); // a little hacky - id is name of property
+        this.updateAttr(eattr, teval);
     }
     // when an input corresponding to a part of an attribute is modified instead of an entire attribute
-    public handleSubChange = (eAttr: string, pathIndex: number, subpathIndex: number, pointIndex: number, event: React.ChangeEvent<HTMLInputElement>) => {
-        // const ptinfo = event.target.id.split("_");
-        // const xory = (ptinfo[2] === "x" ? 0 : 1);
+    public handleMulPtRange = (eAttr: string, pathIndex: number, subpathIndex: number, pointIndex: number, event: React.ChangeEvent<HTMLInputElement>) => {
         const teval = (isNaN(Number(event.target.value))) ? event.target.value : +event.target.value;
-        const path = _.cloneDeep(this.props.eValue.contents);
+        const path = _.cloneDeep(this.props.eValue.contents);   // TODO do we really need to clone it? good in practice but is it strictly necessary??
```

Comment:
`eValue` seems to be a pointer to a property in `state.shapes`? If so, you do want to make a copy when you need to make changes to the value.